### DISCUSSION TITLE
LLVM ERROR: Symbol not found: __svml_cosf8_ha

### DISCUSSION THREAD

**Discussion by samvanity** [erew123/alltalk_tts#87] [Q&A]
Hi,

I tried installing it as part of TextGenWebUi and when I tried to start the extension, I ran into this error message:
LLVM ERROR: Symbol not found: __svml_cosf8_ha

I've done the installation on another computer with TextGenWebUi and everything works there. How do I resolve this issue?

**Comment by erew123** [comment]
Hi @samvanity 

You may need to run the diagnostics so that I can understand your Python environment. I assume your machine is on a recent version/update of Text-generation-webui?

The following **may** resolve it for you:

In Text-generation-webui folder run `cmd_windows.bat`

`pip install librosa==0.9.1` (this will downgrade the librosa to 0.9.1, it will complain about 0.10.1 being required by TTS, but ignore it)

launch the ooba once, then close it

Then once again - run `cmd_windows.bat` and install librosa 0.10.1 again:

`pip install librosa==0.10.1`

Thanks

**Reply by samvanity** [reply]
Thanks! I've actually installed this as a standalone (outside of TextGenWebUi) and it installed and started just fine. 

My next questions - is it possible I change the confignew.json in the AllTalk extension folder of the TextGenWebUi running on computer A to point to a standalone AllTalk server running on computer B?

I tried but it didn't work. The command window on Computer B did receive the text sent to it from the TextGenWebUi running on computer A, but I can't hear the voice on the TextGenWebUi. I set it to use "API TTS" but it still doesn't work.

**Reply by erew123** [reply]
No its not designed for working that way. I'm not exactly clear what you are trying to achieve/what the end setup you are looking for? There is a full API suite available https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-api-suite-and-json-curl where you can interact with a remote AllTalk version (assuming you change its IP address to the external IP address of that computer and post it through your firewall). You can access the AllTalk instance remotely on that IP address too. It depends what you are really trying to achieve.

**Reply by samvanity** [reply]
My computer A has 24G of VRAM (4090) and once the LLM is loaded in TextGenWebUI, I don't have enough VRAM for TTS to be running on the same machine. 

So I installed the extension to get the interface into TextGenWebUI (computer A) and I installed a standalone AllTalk server on Computer B, hoping to connect the TextGenWebUi on computer A to the AllTalk running on computer B by changing the ip in confignew.json to point to computer B. It does send the text from A to B, but the voice doesn't play (computer A does not receive the .wav file).

And by the way, how do I get SillyTavern to show AllTalk in their TTS menu? I see a bunch of other TTS providers but not AllTalk. My SillyTavern and TextGenWebUi are all updated to their most current versions.

**Comment by erew123** [comment]
Hi @samvanity Have you looked in the built in documentation http://127.0.0.1:7851/ at the LowVRAM mode? This is specifically designed for a situation on your local machine where your LLM has filled your VRAM up. It moves the TTS model in and out of VRAM/RAM as required, **temporarily** displacing a couple of layers of the LLM when it does this. Depending on the performance of your PCI bus, this results in usually a maximum 1 second addition to TTS generation (but could be 2 seconds if you have slower PCI and memory). The layers of the LLM move back in when requested and are booted out again when TTS is generated. The built in documentation has more explanation on this process and also you can set this as the default start-up setting at the top of the page. 

![lowvrammode](https://github.com/erew123/alltalk_tts/assets/35898566/922d979d-6b42-4f06-9d0a-af740408b993)
 
Outside of that, the Text-generation-webui interface is looking specifically for a wav file on the local disk, not across a network, which is why its not working in a local/remote setup. It would be possible to write a local/remote text-generation-webui interface, but it doesnt currently support that. 

Would the LowVRAM mode be a suitable solution for you? 

As for SillyTavern, the AllTalk extension/add-in is in their staging/development area currently and hasn't been pushed to the live build https://github.com/SillyTavern/SillyTavern/tree/staging though they did push the documentation live, which obviously leads to a confusing situation. I don't know when they will push it live. 

The current files required for the extension are in `/alltalk_tts/templates/STfiles` and they need to go in `/SillyTavern/public/scripts/extensions/tts` however, if you do use these, its possible you may get a warning when you update SillyTavern in the future, so you may have to remove those files out to update.

Thanks

**Reply by samvanity** [reply]
Thanks a lot for the explanation! I did enable Low VRAM and it worked beautifully. I was thinking about using AllTalk for my all my other TTS needs from a single place (a computer on the same network) so I was experimenting it first with TextGenWebUi. Perhaps storing the wav files on the server with unique ids and streaming them whenever the user requests will work for TextGen too.

And thanks for the explanation about SillyTavern's missing UI. I guess I'll wait until they put it live into the main then.

**Comment by erew123** [comment]
@samvanity Glad youve got it sorted. 

As for the text-gen setup, with a rework, I could make a remote text-gen interface to communicate with a remote AllTalk server. It already stores everything with a unique ID number, so that aspect wouldn't be a challenge. If its something you really think you need, let me know.

Im going to close this for now, but you can still reply on here if needed.

Thanks
